src/textual/widgets/_loading_indicator.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+from time import time
+
+from rich.console import RenderableType
+from rich.style import Style
+from rich.text import Text
+
+from ..color import Gradient
+from ..widget import Widget
+
+
+class LoadingIndicator(Widget):
+    """Display an animated loading indicator."""
+
+    COMPONENT_CLASSES = {"loading-indicator--dot"}
+
+    DEFAULT_CSS = """
+    LoadingIndicator {
+        width: 100%;
+        height: 100%;
+        content-align: center middle;
+    }
+
+    LoadingIndicator > .loading-indicator--dot {
+        color: $accent;
+    }
+
+    """
+
+    def on_mount(self) -> None:
+        self.start_time = time()
```

Comment:
Is this intended to be public? If so it would be a good idea to add a docstring.